packages/lit-dev-content/site/docs/v3/data/task.md

```diff
@@ -0,0 +1,440 @@
+    },
+    args: () => [this.productId]
+  });
+
+  render() {
+    return this._productTask.render({
+      pending: () => html`<p>Loading product...</p>`,
+      complete: (product) => html`
+          <h1>${product.name}</h1>
+          <p>${product.price}</p>
+        `,
+      error: (e) => html`<p>Error: ${e}</p>`
+    });
+  }
+}
+```
+
+{% endswitchable-sample %}
+
+### Features
+
+Task takes care of a number of things needed to properly manage async work:
+- Gathers task arguments when the host updates
+- Runs task functions when arguments change
+- Tracks the task status (initial, pending, complete, or error)
+- Saves the last completion value or error of the task function
+- Triggers a host update when the task changes status
+- Handles race conditions, ensuring that only the latest task invocation completes the task
+- Renders the correct template for the current task status
+- Allows aborting tasks with an `AbortController`
+
+This removes most of the boilerplate for correctly using async data from your code, and ensures robust handling of race conditions and other edge-cases.
+
+## What is async data?
+
+Async data is data that's not available immediately, but may be available at some time in the future. For example, instead of a value like a string or an object that's usable synchronously, a Promise provides a value in the future.
+
+Async data is usually returned from an async API, which can come in a few forms:
+- Promises and async functions which return Promises
+- Functions that accept callbacks
+- Objects that emit events
+- Various async patterns libraries like observables and signals
+
+## What is a task?
+
+At the core of the Task controller is the concept of a "task" itself.
+
+A task is an async operation which does some work to produce data. A task can be in a few different states (initial, pending, complete, and error) and can take parameters.
+
+A task is a generic concept and could represent any async operation. They apply best when there is a request/response structure, such as a network fetch, database query, or waiting for a single event in response to some action. They're less applicable to spontaneous or streaming operations like an open-ended stream of events, a streaming database response, etc.
+
+## Installation
+
+```bash
+npm i @lit/task
+```
+
+## Usage
+
+`Task` is a [reactive controller](../../composition/controllers/), so it can respond to and trigger updates to Lit's reactive update lifecycle.
+
+You'll generally have one Task object for each logical task that your component needs to perform. Install tasks as fields on your class:
+
+{% switchable-sample %}
+
+```ts
+class MyElement extends LitElement {
+  private _myTask = new Task(this, {/*...*/});
+}
+```
+
+```js
+class MyElement extends LitElement {
+  _myTask = new Task(this, {/*...*/});
+}
+```
+
+{% endswitchable-sample %}
+
+
+As a class field, the task status and value are easily available:
+
+```ts
+this._task.value
```

Comment:
Nit: in sentence above introduce `status`, and `value`. Could this example also have `this._task.status`?